pkg/secretinformer/informer.go

```diff
@@ -0,0 +1,85 @@
+package secretinformer
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+)
+
+const resyncTime = 10 * time.Minute
+
+// SecretInformer is a convenience wrapper around a Kubernetes informer for a specific secret.
+type SecretInformer struct {
```

Comment:
nit: adding a interface implementation check for `ResourceEventHandler` would be nice, easier to fix things in case upgrade of k8s libs requires changes to the event handler.